src/Dotnet.Status.Web/Controllers/AlertHookController.cs

```diff
@@ -0,0 +1,325 @@
+        {
+            string org = _githubOptions.Value.Organization;
+            string repo = _githubOptions.Value.Repository;
+            _logger.LogInformation(
+                "Alert state detected for {ruleUrl} in stage {ruleState}, porting to github repo {org}/{repo}",
+                notification.RuleUrl,
+                notification.State,
+                org,
+                repo);
+
+            IGitHubClient client =
+                await GetGitHubClientAsync(_githubOptions.Value.Organization, _githubOptions.Value.Repository);
+            Issue issue = await GetExistingIssueAsync(client, notification);
+            await EnsureLabelsAsync(client, org, repo);
+            if (issue == null)
+            {
+                _logger.LogInformation("No existing issue found, creating new active issue with {label}",
+                    ActiveAlertLabel);
+                issue = await client.Issue.Create(org, repo, GenerateNewIssue(notification));
+                _logger.LogInformation("Github issue {org}/{repo}#{issueNumber} created", org, repo, issue.Number);
+            }
+            else
+            {
+                _logger.LogInformation(
+                    "Found existing issue {org}/{repo}#{issueNumber}, replacing {inactiveTag} with {activeTag}",
+                    org,
+                    repo,
+                    issue.Number,
+                    InactiveAlertLabel,
+                    ActiveAlertLabel);
+
+                await TryRemove(() => client.Issue.Labels.RemoveFromIssue(org, repo, issue.Number, InactiveAlertLabel));
+                await TryCreate(() =>
+                    client.Issue.Labels.AddToIssue(org, repo, issue.Number, new[] {ActiveAlertLabel}));
+
+                _logger.LogInformation("Adding recurrence comment to  {org}/{repo}#{issueNumber}",
+                    org,
+                    repo,
+                    issue.Number);
+                IssueComment comment = await client.Issue.Comment.Create(org,
+                    repo,
+                    issue.Number,
+                    GenerateNewNotificationComment(notification));
+                _logger.LogInformation("Created comment {org}/{repo}#{issue}-issuecomment-{comment}",
+                    org,
+                    repo,
+                    issue.Id,
+                    comment.Id);
+            }
+        }
+
+        private string GenerateNewNotificationComment(GrafanaNotification notification)
+        {
+            var metricText = new StringBuilder();
+            foreach (GrafanaNotificationMatch match in notification.EvalMatches)
+            {
+                metricText.AppendLine($"  - *{match.Metric}* {match.Value}");
+            }
+            
+            string icon = GetIcon(notification);
+
+            return $@":{icon}: Metric state changed to *{notification.State}*
+
+> {notification.Message.Replace("\n", "\n> ")}
+
+{metricText}
+
+![Metric Graph]({notification.ImageUrl})
```

Comment:
This reminds me: We should check that the time period before image clean-up is sufficient. The docs say it defaults to 24 hours, but all of the test alerts over the last month are still shown.

Reply:
Do you have a reference to said doc?  That would be really lame if the images weren't good for more than a day.  But whether it is or not, we'll still want to pop this here.  Even if it gets cleaned up, the issue is still useful, just missing the cool looking graph.

---
In reply to: [344329076](https://github.com/dotnet/arcade-services/pull/756#discussion_r344329076) [](ancestors = 344329076)

Reply:
[Image Rendering](https://grafana.com/docs/administration/image_rendering/#image-rendering) and the (temp-data-lifetime)[https://grafana.com/docs/installation/configuration/#temp-data-lifetime] configuration setting but I see now that this is the step before upload and really is just temporary. I don't see any information about clean-up of real data so probably no action needed.